source/agora/consensus/protocol/Nominator.d

```diff
@@ -324,17 +323,22 @@ extern(D):
 
     private void checkNominate () @safe
     {
+        // already in the asynchronous stage of nominating
+        if (this.is_nominating)
+            return;
+
         const cur_time = this.clock.networkTime();
-        if (cur_time < this.params.GenesisStartTime)
+        const genesis_start_time = this.params.Genesis.header.timestamp;
+        if (cur_time < genesis_start_time)
         {
             log.fatal("Clock is out of sync. " ~
                 "Current time: {}. Genesis time: {}", cur_time,
-                this.params.GenesisStartTime);
+                genesis_start_time);
             return;
         }
 
         const slot_idx = this.ledger.getBlockHeight() + 1;
```

Comment:
Could you move this line above `this.nominate(slot_idx, data);`? Because the `slot_idx` is not used in the next line anymore.